android/src/main/java/nordpol/TagDispatcher.java

```diff
@@ -0,0 +1,95 @@
+package nordpol.android;
+
+import android.nfc.Tag;
+import android.nfc.NfcAdapter;
+import android.nfc.tech.IsoDep;
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.app.PendingIntent;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Build;
+import android.os.Bundle;
+
+public class TagDispatcher {
+    private NfcAdapter.ReaderCallback callback;
+    private Activity activity;
+    private NfcAdapter adapter;
+
+    private TagDispatcher(Activity activity, NfcAdapter adapter,
+                          NfcAdapter.ReaderCallback callback) {
+        this.activity = activity;
+        this.adapter = adapter;
+        this.callback = callback;
+    }
+
+    public static TagDispatcher get(Activity activity, NfcAdapter adapter,
+                                    NfcAdapter.ReaderCallback callback) {
+        return new TagDispatcher(activity, adapter, callback);
+    }
+
+    @TargetApi(Build.VERSION_CODES.KITKAT)
+    public void enableExclusiveNfc() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            enableReaderMode();
+        } else {
+            enableForegroundDispatch();
+        }
+    }
+
+    @TargetApi(Build.VERSION_CODES.KITKAT)
+    public void disableExclusiveNfc() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            disableReaderMode();
+        } else {
+            disableForegroundDispatch();
+        }
+    }
+
+    public void interceptIntent(Intent intent) {
```

Comment:
I am wondering whether we should add some mean to signal if the intent was intercepted (meaning already processed)